components/core/src/clp/streaming_archive/writer/Archive.cpp

```diff
@@ -647,6 +637,19 @@ void Archive::update_metadata() {
     }
 }
 
+auto Archive::update_global_metadata() -> void {
+    m_global_metadata_db->open();
+    if (false == m_local_metadata.has_value()) {
+        throw OperationFailed(ErrorCode_Failure, __FILENAME__, __LINE__);
+    }
+    m_global_metadata_db->add_archive(m_id_as_string, m_local_metadata.value());
+    m_global_metadata_db->update_metadata_for_files(m_id_as_string, m_files_written);
+    for (auto* file : m_files_written) {
+        delete file;
+    }
```

Comment:
Just looking at the function name, one may not expect the function to call delete on m_files_written.

I feel it would be more clear to put these 3 lines out side of the function, or maybe update the funciton name to include the "delete" part